src/main.rs

```diff
@@ -21,9 +22,38 @@ fn main() -> Result<(), Box<dyn Error>> {
     let host = matches.get_one::<String>("Broker").unwrap();
     let port = *matches.get_one::<u16>("Port").unwrap();
 
-    let client_id = format!("mqttui-{:x}", rand::random::<u32>());
+    let client_id = matches
+        .get_one::<String>("ClientId")
+        .cloned()
+        .unwrap_or_else(|| format!("mqttui-{:x}", rand::random::<u32>()));
+
+    let encryption = matches.get_one::<bool>("Encryption").copied();
+    let encryption = match (encryption, port) {
+        (Some(encryption), _) => encryption,
+        (None, 8883) => true,
+        _ => false,
+    };
+
     let mut mqttoptions = MqttOptions::new(client_id, host, port);
     mqttoptions.set_max_packet_size(usize::MAX, usize::MAX);
+    if encryption {
+        let certs = rustls_native_certs::load_native_certs().unwrap();
+        let mut roots = rustls::RootCertStore::empty();
+        for cert in certs {
+            let _e = roots.add(&rustls::Certificate(cert.0));
```

Comment:
I think it should either use `drop(…)` or handle that error somehow?

Reply:
well, there are non working certificates in many distributions. I think also that the certificate parsing is not supporting all cases yet. 

with this in mind, just silently ignoring the malfunctioning certificates seems ok to me, and seems to be what other tools do.

Reply:
Alright then. `drop(…)` is annoyed here but I would remove the e here as we dont need a name anyway.

```suggestion
            let _ = roots.add(&rustls::Certificate(cert.0));
```